README.md

```diff
@@ -58,81 +54,52 @@ Include this badge in your readme if you make a new module that implements inter
 
 A valid (read: that follows this interface) IPLD format implementation the following API.
 
-### IPLD format utils
+IPLD Format APIs are restricted to a single IPLD Node, they never access any linked IPLD Nodes.
 
-#### `util.serialize(dagNode, callback)`
 
-> serializes a dagNode of an IPLD format into its binary format
+### `serialize(IpldNode)`
 
-`callback` must have the signature `function (err, binaryBlob)`, where `err` is an Error is the function fails and `binaryBlob` is a Buffer containing the serialized version.
+> Serializes the internal representation of an IPLD Node into a binary blob.
 
-#### `util.deserialize(binaryBlob, callback)`
+`IpldNode` is a previously deserialized binary Block.
 
-> deserializes a binary blob into the instance
+Returns a Promise containing a `Buffer` with the serialized version of the given IPLD Node.
```

Comment:
I would really encourage to remove requirement to return Buffer in favor of Uint8Array. This requirement makes browser use case unnecessarily complicated, and in practice isn't really necessary. Other IPFS libraries can do the casting where necessary.

P.S. node Buffer is also a valid Uint8Array so node case will require no change.

Reply:
I would love if there was some word about errors as well. Specifically how decoder can return error and how that error propagates.

Reply:
Moving from Node.js to Browser based primitives is certainly on my todo list, but out of scope for these API changes. I don't want to do everything at once.